agi.go

```diff
@@ -243,6 +243,11 @@ func (a *AGI) Command(cmd ...string) (resp *Response) {
 			break
 		}
 
+		if strings.HasPrefix(raw, "HANGUP") {
+			resp.Error = fmt.Errorf("channel hung up")
```

Comment:
I think I would sentinel error this.  That is, create a

```go
// ErrHangup indicates the channel hung up during processing
var ErrHangup = errors.New("hangup")
```

Then here:

```go
resp.Err = ErrHangup
```

This just makes it cleaner for checking this kind of thing downstream.  Hangup errors are frequently treated differently from other types of errors.